net2/SecondaryInterface.js

```diff
@@ -104,17 +107,16 @@ exports.create = async function(config) {
   for (const intf of list) {
     const subnets = getSubnets(intf.name, 'IPv4');
 
-    const overlapped = subnets.find(net =>
-      net.contains(secondarySubnet.firstAddress) ||
-      net.contains(secondarySubnet.lastAddress) ||
-      secondarySubnet.contains(net.firstAddress) ||
-      secondarySubnet.contains(net.lastAddress)
-    )
+    let overlapped = subnets.includes(secondaryIpSubnet);
+
+    const sysManager = new SysManager();
+    if (secondaryIpSubnet.split('/')[0] === sysManager.myGateway())
+      overlapped = true;
```

Comment:
might be a good idea to separate the logic between manual setup and auto setup.
basic check or no check for manual setup
verbose check for auto setup